apps/mobile/src/screens/HomeScreen/CuratedScreen.tsx

```diff
@@ -73,15 +83,32 @@ function CuratedScreenInner({ queryRef }: CuratedScreenInnerProps) {
     return [...curatedEvents];
   }, [curatedFeed?.edges]);
 
+  const [showWelcome, setShowWelcome] = useState(false);
+
+  const checkShouldShowWelcome = useCallback(async () => {
+    const shown = await AsyncStorage.getItem('welcomeMessageShown');
+    if (shown !== 'true') {
+      setShowWelcome(true);
+      await AsyncStorage.setItem('welcomeMessageShown', 'true');
+    }
+  }, [setShowWelcome]);
+
+  useEffect(() => {
+    checkShouldShowWelcome();
+  }, [checkShouldShowWelcome]);
```

Comment:
nit: this looks like an extremely useful hook to make generic, like [usePersistedState](https://github.com/gallery-so/gallery/blob/main/apps/web/src/hooks/usePersistedState.ts)

Reply:
yesss True!! I think Rohan is adding one in his PR for debugger mode , do you think it's chill to wait to switch it out until that's merged? We already have this code in prod (just moving files) so should be minimal downside

Reply:
💯